db/l2db/l2db.go

```diff
@@ -147,58 +149,107 @@ func (l2db *L2DB) UpdateTxsInfo(txs []common.PoolL2Tx, batchNum common.BatchNum)
 	return nil
 }
 
-// NewPoolL2TxWriteFromPoolL2Tx creates a new PoolL2TxWrite from a PoolL2Tx
-func NewPoolL2TxWriteFromPoolL2Tx(tx *common.PoolL2Tx) *PoolL2TxWrite {
-	// transform tx from *common.PoolL2Tx to PoolL2TxWrite
-	insertTx := &PoolL2TxWrite{
-		TxID:      tx.TxID,
-		FromIdx:   tx.FromIdx,
-		TokenID:   tx.TokenID,
-		Amount:    tx.Amount,
-		Fee:       tx.Fee,
-		Nonce:     tx.Nonce,
-		State:     common.PoolL2TxStatePending,
-		Signature: tx.Signature,
-		RqAmount:  tx.RqAmount,
-		Type:      tx.Type,
-	}
-	if tx.ToIdx != 0 {
-		insertTx.ToIdx = &tx.ToIdx
+// AddTxTest inserts a tx into the L2DB, without security checks. This is useful for test purposes,
+func (l2db *L2DB) AddTxTest(tx *common.PoolL2Tx) error {
+	// Add tx without checking if pool is full
+	_, err := l2db.addTx(tx, false)
+	return err
+}
+
+func (l2db *L2DB) addTx(tx *common.PoolL2Tx, checkPoolIsFull bool) (sql.Result, error) {
+	// Prepare extra DB fields and nullables
+	// AmountFloat
+	f := new(big.Float).SetInt((*big.Int)(tx.Amount))
+	amountF, _ := f.Float64()
+	// ToIdx
+	// var toIdx *common.Idx
+	// if tx.ToIdx != 0 {
+	// 	toIdx = new(common.Idx)
+	// 	*toIdx = tx.ToIdx
+	// }
+	// ToEthAddr
+	var toEthAddr *ethCommon.Address
+	if tx.ToEthAddr != common.EmptyAddr {
+		toEthAddr = new(ethCommon.Address)
+		*toEthAddr = tx.ToEthAddr
 	}
-	nilAddr := ethCommon.BigToAddress(big.NewInt(0))
-	if tx.ToEthAddr != nilAddr {
-		insertTx.ToEthAddr = &tx.ToEthAddr
+	// ToBJJ
+	var toBJJ *babyjub.PublicKeyComp
+	if tx.ToBJJ != common.EmptyBJJComp {
+		toBJJ = new(babyjub.PublicKeyComp)
+		*toBJJ = tx.ToBJJ
 	}
+	// Info (always nil)
+	var info *string
+	// Rq fields, nil unless tx.RqFromIdx != 0
+	var rqFromIdx *common.Idx
+	var rqToIdx *common.Idx
+	var rqToEthAddr *ethCommon.Address
+	var rqToBJJ *babyjub.PublicKeyComp
+	var rqTokenID *common.TokenID
+	var rqAmount *string
+	var rqFee *common.FeeSelector
+	var rqNonce *common.Nonce
 	if tx.RqFromIdx != 0 {
-		insertTx.RqFromIdx = &tx.RqFromIdx
-	}
-	if tx.RqToIdx != 0 { // if true, all Rq... fields must be different to nil
-		insertTx.RqToIdx = &tx.RqToIdx
-		insertTx.RqTokenID = &tx.RqTokenID
-		insertTx.RqFee = &tx.RqFee
-		insertTx.RqNonce = &tx.RqNonce
-	}
-	if tx.RqToEthAddr != nilAddr {
-		insertTx.RqToEthAddr = &tx.RqToEthAddr
-	}
-	if tx.ToBJJ != common.EmptyBJJComp {
-		insertTx.ToBJJ = &tx.ToBJJ
+		// RqFromIdx
+		rqFromIdx = new(common.Idx)
+		*rqFromIdx = tx.RqFromIdx
```

Comment:
```suggestion
		rqFromIdx = &tx.RqFromIdx
```
Maybe is it possible to create it like this and don't create a redundant assignment? Applicable to all the following assignments after this.